pages/plants/[id]/index.js

```diff
@@ -5,20 +5,44 @@ import { useRouter } from "next/router";
 import styled from "styled-components";
 import useSWR from "swr";
 
+const lightNeedMap = {
+  1: "⛅",
+  2: "🌤️",
+  3: "☀️",
+};
+
+const waterNeedMap = {
+  1: "💧",
+  2: "💧💧",
+  3: "💧💧💧",
+};
+
+const seasonMap = {
+  spring: "🌸 Spring",
+  summer: "☀️ Summer",
+  autumn: "🍂 Autumn",
+  winter: "❄️ Winter",
+};
+
 export default function DetailsPage() {
   const router = useRouter();
   const { isReady } = router;
   const { id } = router.query;
 
   const { data: plant, isLoading, error } = useSWR(`/api/plants/${id}`);
 
-  if (!isReady || isLoading || error || !plant) return <h2>Loading...</h2>;
+  if (isLoading || !isReady) {
+    return <h2>Loading...</h2>;
+  }
+  if (error || !plant) {
+    return <h2>Error loading plant data</h2>;
+  }
 
   const seasons = plant.fertiliserSeason;
 
   return (
     <>
-      <BackButton />
+      <BackButton href="/" />
```

Comment:
I think you don't need to pass the href here, since this is already your default value